apps/wallet-api-tools/src/components/GroupedMessage.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState } from "react";
+import { MessageGrouped } from "../types";
+import { getDate } from "../helpers";
+import { coolGlow, dracula } from "thememirror";
+import Message from "./Message";
+
+type Props = {
+  group: MessageGrouped["messages"];
+  setValue: (value: string) => void;
+  displayModal?: boolean;
+};
+
+const GroupedMessage = ({ group, setValue, displayModal }: Props) => {
+  const parsedSentValue = JSON.parse(group.sent.value) as {
+    id: string;
+    jsonrpc: string;
+    method: string;
+    [value: string]: string;
+  };
+  const [displayResponse, setDisplayResponse] = useState(displayModal ?? false);
+
+  return (
+    <div className="w-full  h-[max-content] rounded-md">
+      <div className="text-lg">
+        <p>Method : {parsedSentValue.method}</p>
+      </div>
+      <p className="text-slate-400 text-xs mb-2">{getDate(group.sent)}</p>
+      <div className=" max-w-full">
+        <div
+          className={`max-w-full h-[max-content] bg-[#060521] p-4 pr-12 rounded-xl border-solid border-[1px] border-[#666]`}
+        >
+          <div className="flex gap-x-2">
+            <button
+              onClick={() => {
+                // eslint-disable-next-line @typescript-eslint/no-unused-vars
+                const { id, jsonrpc, ...rest } = parsedSentValue;
+                setValue(JSON.stringify(rest, null, 2));
+              }}
```

Comment:
We should probably have a `onClear` prop in this component and delegate the logic to the parent